frontend/web/components/segments/Rule/components/RuleConditionValueInput.tsx

```diff
@@ -18,19 +24,69 @@ type RuleConditionValueInputProps = {
   readOnly?: boolean
   isValid?: boolean
   projectId?: number
-  showEnvironmentDropdown?: boolean
-  operator?: string
+  operator?: OperatorValue
+  property?: string
+  className?: string
 }
 
 const RuleConditionValueInput: React.FC<RuleConditionValueInputProps> = ({
   isValid,
   onChange,
   operator,
+  property,
   projectId,
-  showEnvironmentDropdown,
   value,
+  className,
   ...props
 }) => {
+  const { data } = useGetEnvironmentsQuery(
+    { projectId: projectId?.toString() || '' },
+    { skip: !projectId },
+  )
+  const environmentOptions = useMemo(
+    () =>
+      data?.results
+        ? data.results.map(({ name }) => ({ label: name, value: name }))
+        : [],
+    [data],
+  )
+
+  const { showMultiEnvironmentSelect, showSingleEnvironmentSelect } = useConditionInputType(
+    operator,
+    property,
+  )
+
+  if (showMultiEnvironmentSelect) {
+    return (
+      <div className={className}>
+        <MultiSelect
+          selectedValues={safeParseArray(value)}
+          onSelectionChange={(selectedValues: string[]) => {
+            onChange?.(JSON.stringify(selectedValues))
+          }}
```

Comment:
Critical: We have a payload formatting issue here. It might be also related to a backend issue. We want to accept JSON-encoded lists. See this [issue](https://github.com/Flagsmith/flagsmith/issues/5935)

However right now it is not working. You can test it by:
- Creating a segment with ONE Rule environment.name IN with one or more environment 
- Go to Users and select one of the environment
- Users should be within the segment (it's not even with one env.

If you checkout to main and use the normal csv format for `IN` it will work

https://www.loom.com/share/faca8617e13b4338bbaf0a263ef00f25